gnocchi/metric/v1/resources/requests.go

```diff
@@ -132,3 +132,48 @@ func Create(client *gophercloud.ServiceClient, resourceType string, opts CreateO
 	})
 	return
 }
+
+// UpdateOptsBuilder allows extensions to add additional parameters to the
+// Update request.
+type UpdateOptsBuilder interface {
+	ToResourceUpdateMap() (map[string]interface{}, error)
+}
+
+// UpdateOpts represents options used to update a network.
+type UpdateOpts struct {
+	// Metrics field can be used to link existing metrics in the resource
+	// or to create metrics and update the resource at the same time to save
+	// some requests.
+	Metrics map[string]interface{} `json:"metrics,omitempty"`
```

Comment:
In order to delete a metric from a resource, Gnocchi wants an empty `metrics` object:

```
REQ: curl -g -i -X PATCH http://127.0.0.1:8041/v1/resource/generic/fba056ff-60f3-4edc-bb17-7c84531fbb53 -H "User-Agent: gnocchi keystoneauth1/3.1.0 python-requests/2.14.2 CPython/2.7.5" -H "Content-Type: application/json" -H "Accept: application/json, */*" -H "X-Auth-Toke
n: {SHA1}122e9eefd8a4fc1ead3e2cbce2b564b7c147d175" -d '{"metrics":{}}'
```

In order to do this, we need to change the above to:

```go
Metrics *map[string]interface{} `json:"metrics,omitempty"`
```

This way metrics can be deleted by specifying a pointer to a zero-value map. In addition, not specifying a `Metrics` field will not inadvertently cause metrics to be deleted.